daphne_worker/src/durable/reports_processed.rs

```diff
@@ -53,18 +54,69 @@ pub struct ReportsProcessed {
     alarmed: bool,
 }
 
-impl ReportsProcessed {
-    /// Check if the report has been processed. If not, return None; otherwise, return the ID.
-    async fn to_checked(&self, report_id: ReportId) -> Result<Option<ReportId>> {
-        let key = format!("processed/{}", report_id.to_hex());
-        let processed: bool = state_set_if_not_exists(&self.state, &key, &true)
-            .await?
-            .unwrap_or(false);
-        if processed {
-            Ok(Some(report_id))
-        } else {
-            Ok(None)
+#[derive(Debug, Clone)]
+struct ReportIdKey<'s>(&'s ReportId, String);
+
+impl<'id> From<&'id ReportId> for ReportIdKey<'id> {
+    fn from(id: &'id ReportId) -> Self {
+        ReportIdKey(id, format!("processed/{}", id.to_hex()))
+    }
+}
+
+#[derive(Debug)]
+enum CheckedReplays<'s> {
+    SomeReplayed(Vec<&'s ReportId>),
+    AllFresh(Vec<ReportIdKey<'s>>),
+}
```

Comment:
I believe this works as an alternative `Default` implementation
```suggestion
#[derive(Debug, Default)]
enum CheckedReplays<'s> {
    SomeReplayed(Vec<&'s ReportId>),
    #[default]
    AllFresh(Vec<ReportIdKey<'s>>),
}
```

Reply:
sadly it doesn't work, it only works for Unit variants

Reply:
:(